packages/node/src/fs.js

```diff
@@ -0,0 +1,148 @@
+ * real file
+ *
+ * @param {PathLike} filepath Path to check
+ * @param {WithFs} [options] Options
+ * @returns {boolean}
+ */
+export const isFileSync = (filepath, { fs = nodeFs } = {}) => {
+  return !!fs.statSync(filepath, { throwIfNoEntry: false })?.isFile()
+}
+
+/**
+ * Returns `true` if the file at `filepath` is readable
+ *
+ * @param {PathLike} filepath
+ * @param {WithFs} [options]
+ * @returns {boolean}
+ */
+export const isReadablePathSync = (filepath, { fs = nodeFs } = {}) => {
+  try {
+    fs.accessSync(filepath, fs.constants.R_OK)
+    return true
+  } catch {
+    return false
+  }
+}
+
+/**
+ * Returns `true` if the file at `filepath` is executable (and readable)
+ *
+ * @param {PathLike} filepath Path to check
+ * @param {WithFs} [options] Options
+ * @returns {boolean}
+ */
+export const isExecutablePathSync = (filepath, { fs = nodeFs } = {}) => {
+  try {
+    fs.accessSync(filepath, fs.constants.X_OK & fs.constants.R_OK)
+    return true
+  } catch {
+    return false
+  }
+}
+
+/**
+ * Returns `true` if the file at `filepath` is both a real file _and_ readable
+ *
+ * @param {PathLike} filepath Path to check
+ * @param {WithFs} [options] Options
+ * @returns {boolean}
+ */
+export const isReadableFileSync = (filepath, options) => {
+  return isFileSync(filepath, options) && isReadablePathSync(filepath, options)
+}
+
+/**
+ * Returns `true` if the file at `filepath` is both a symlink _and_ executable
+ * (and readable)
+ *
+ * @param {PathLike} filepath Path to check
+ * @param {WithFs} [options] Options
+ * @returns {boolean}
+ */
+export const isExecutableSymlink = (filepath, options) => {
+  return (
+    isSymlinkSync(filepath, options) && isExecutablePathSync(filepath, options)
+  )
+}
+
+/**
+ * Returns `true` if the file at `filepath` is a symlink
+ *
+ * @param {PathLike} filepath Path to check
+ * @param {WithFs} [options] Options
+ * @returns {boolean}
+ */
+export const isSymlinkSync = (filepath, { fs = nodeFs } = {}) => {
+  return !!fs.lstatSync(filepath, { throwIfNoEntry: false })?.isSymbolicLink()
+}
+
+/**
+ * Asserts a path is absolute
+ *
+ * @param {string} filepath Path to check
+ * @param {string} [failureMessage] Message to include in the error if the path
+ *   is not absolute
+ * @returns {void}
+ * @throws `AssertionError`
+ */
+export const assertAbsolutePath = (filepath, failureMessage) => {
+  assert(isAbsolutePath(filepath), failureMessage)
+}
+
+/**
+ * Returns `true` if `filepath` is absolute
+ *
+ * @param {string} filepath Path to check
+ * @returns {boolean}
+ */
+export const isAbsolutePath = (filepath) => {
+  return path.isAbsolute(filepath)
+}
+
+/**
+ * Returns the "real" path of `filepath`
+ *
+ * @param {PathLike} filepath Path to check
+ * @param {WithFs} options Options
+ * @returns {string}
+ */
+export const realpathSync = (filepath, { fs = nodeFs } = {}) => {
+  // `realpath` can return a `Buffer` if you provide that as the desired encoding,
+  // which is weird?
```

Comment:
it could be to avoid character encoding issues on different systems
when special or non-ASCII chars are in path

Reply:
makes sense, yeah, for cases where conversion would be lossy.